File: FRONTEND/hero-pricing-engine-ui/src/services/configurationService.js
import api from "../api/api";

export const getConfigurations=()=>{

    return api.get("/configurations");

}

export const createConfiguration=(configuration)=>{

    return api.post("/configurations",configuration);

}

export const getConfiguration=(id)=>{

    return api.get(`/configurations/${id}`);

}

export const deleteConfiguration=(id)=>{

    return api.delete(`/configurations/${id}`);

}

File: BACKEND/PricingEngineApplication/src/main/java/com/example/PricingEngineApplication/service/ConfigurationService.java
package com.example.PricingEngineApplication.service;


import java.util.List;

import com.example.PricingEngineApplication.entity.CycleConfiguration;

public interface ConfigurationService {

    CycleConfiguration createConfiguration(CycleConfiguration configuration);

    CycleConfiguration getConfiguration(Long id);

    List<CycleConfiguration> getAllConfigurations();

    void deleteConfiguration(Long id);
}
